app/core/database.py

```diff
@@ -0,0 +1,68 @@
+import contextlib
+from typing import AsyncIterator
+
+from sqlalchemy.ext.asyncio import (
+    AsyncConnection,
+    AsyncEngine,
+    AsyncSession,
+    async_sessionmaker,
+    create_async_engine,
+)
+from sqlmodel import SQLModel
+
+
+class DatabaseSessionManager:
```

Comment:
We definitely need documentation here - how to use this and what model of session management you get

Reply:
added